### PR TITLE
Export support for metrics view comparison toplist

### PR DESCRIPTION
**Pull request by pjain1**

### CODE REVIEW

File: runtime/queries/metricsview_comparison_toplist.go [hunk 1]
```diff
@@ -545,7 +546,159 @@ func (q *MetricsViewComparisonToplist) buildMetricsComparisonTopListSQL(mv *runt
...
+			}
+			meta[4+i*4] = &runtimev1.MetricsViewColumn{
+				Name: fmt.Sprintf("%s_delta_rel", m.MeasureName),
+			}
+		}
+	} else {
+		for i, m := range q.Result.Rows[0].MeasureValues {
+			meta[1+i] = &runtimev1.MetricsViewColumn{
+				Name: fmt.Sprintf("%s_value", m.MeasureName),
+			}
```
Comment (begelundmuller) [inline runtime/queries/metricsview_comparison_toplist.go:634]: Can also just use `m.MeasureName` directly

File: runtime/queries/metricsview_comparison_toplist.go [hunk 2]
```diff
@@ -545,7 +546,159 @@ func (q *MetricsViewComparisonToplist) buildMetricsComparisonTopListSQL(mv *runt
...
+			Fields: map[string]*structpb.Value{
+				q.DimensionName: {
+					Kind: &structpb.Value_StringValue{
+						StringValue: row.DimensionValue.GetStringValue(),
+					},
+				},
+			},
+		}
+		for _, m := range row.MeasureValues {
+			if !isTimeRangeNil(q.ComparisonTimeRange) {
```
Comment (begelundmuller) [inline runtime/queries/metricsview_comparison_toplist.go:650]: Given the number of iterations, maybe pre-compute this instead of calling every time?

**Review by begelundmuller — CHANGES_REQUESTED** [review]

File: runtime/queries/metricsview_comparison_toplist.go [hunk 3]
```diff
@@ -545,7 +546,159 @@ func (q *MetricsViewComparisonToplist) buildMetricsComparisonTopListSQL(mv *runt
...
+}
+
+func (q *MetricsViewComparisonToplist) generateFilename(mv *runtimev1.MetricsView) string {
+	filename := strings.ReplaceAll(mv.Model, `"`, `_`)
+	filename += "_" + q.DimensionName
+	if q.Filter != nil && (len(q.Filter.Include) > 0 || len(q.Filter.Exclude) > 0) {
+		filename += "_filtered"
+	}
+	return filename
 }
```
[the remark this answers is not in the document]
Comment (pjain1) [reply runtime/queries/metricsview_comparison_toplist.go:702]: all export api uses this name, should I change at all places ?

File: runtime/queries/metricsview_comparison_toplist.go [hunk 4]
```diff
@@ -545,7 +546,159 @@ func (q *MetricsViewComparisonToplist) buildMetricsComparisonTopListSQL(mv *runt
...
+				if err != nil {
+					return fmt.Errorf("error building query: %w", err)
+				}
+			}
+
+			filename := q.generateFilename(q.MetricsView)
+			if err := duckDBCopyExport(ctx, w, opts, sql, args, filename, olap, opts.Format); err != nil {
+				return err
+			}
+		} else {
```
[the remark this answers is not in the document]
Comment (pjain1) [reply runtime/queries/metricsview_comparison_toplist.go:576]: FYI, I have observed that duckdb exports does not output headers. Not sure I understand your comment about separate function.

File: runtime/queries/metricsview_comparison_toplist.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (pjain1) [reply runtime/queries/metricsview_comparison_toplist.go]: changed at all places

File: runtime/queries/metricsview_comparison_toplist.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (begelundmuller) [reply runtime/queries/metricsview_comparison_toplist.go:576]: For headers, it was a bug – merge main to fix: https://github.com/rilldata/rill/pull/3117

For separate functions, probably not needed if keeping the deltas. We can probably just add appropriate `... AS name`  to the queries.

File: runtime/queries/metricsview_comparison_toplist.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (begelundmuller) [reply runtime/queries/metricsview_comparison_toplist.go]: Not for metrics view rows (since it accesses the underlying model)

**Review by begelundmuller — APPROVED** [review]

**Pull request merged**